server/src/test/java/org/opensearch/index/shard/IndexShardTests.java

```diff
@@ -4957,4 +4958,39 @@ private static void assertRemoteSegmentStats(
         assertTrue(remoteSegmentStats.getTotalRejections() > 0);
         assertEquals(remoteSegmentTransferTracker.getRejectionCount(), remoteSegmentStats.getTotalRejections());
     }
+
+    public void testShouldRefreshOnTooManyRemoteTranslogFiles() throws Exception {
```

Comment:
does it fail w/o these changes ?

Reply:
Yes, it will fail as `primaryShard.afterWriteOperation();` will not refresh the shard, and hence searchable doc count will always be 0.